File: src/sys/dotnet/fan/sys/Endian.cs
//
// Copyright (c) 2009, Brian Frank and Andy Frank
// Licensed under the Academic Free License version 3.0
//
// History:
//   18 Dec 09  Brian Frank  Creation
//

namespace Fan.Sys
{
  /// <summary>
  /// Endian.
  /// </summary>
  public sealed class Endian : Enum
  {

    public static readonly Endian m_big    = new Endian(0, "big");
    public static readonly Endian m_little = new Endian(1, "little");

    internal static readonly Endian[] array =
    {
      m_big, m_little
    };

    public static readonly List m_vals = new List(Sys.EndianType, array).ro();

    private Endian(int ordinal, string name)
    {
      Enum.make_(this, ordinal, System.String.Intern(name));
    }

    public static Endian fromStr(string name) { return fromStr(name, true); }
    public static Endian fromStr(string name, bool check)
    {
      return (Endian)doFromStr(Sys.EndianType, name, check);
    }

    public override Type @typeof() { return Sys.EndianType; }
 }
}

File: src/sys/java/fan/sys/Endian.java
//
// Copyright (c) 2009, Brian Frank and Andy Frank
// Licensed under the Academic Free License version 3.0
//
// History:
//   18 Dec 09  Brian Frank  Creation
//
package fan.sys;

/**
 * Endian
 */
public final class Endian
  extends Enum
{

  public static final int BIG    = 0;
  public static final int LITTLE = 1;

  public static final Endian big    = new Endian(BIG, "big");
  public static final Endian little = new Endian(LITTLE, "little");

  static final Endian[] array = { big, little };

  public static final List<Endian> vals = (List)new List(Sys.EndianType, array).toImmutable();

  private Endian(int ordinal, String name)
  {
    Enum.make$(this, FanInt.pos[ordinal], name.intern());
  }

  public static Endian fromStr(String name) { return fromStr(name, true); }
  public static Endian fromStr(String name, boolean checked)
  {
    return (Endian)doFromStr(Sys.EndianType, name, checked);
  }

  public Type typeof() { return Sys.EndianType; }

}

